compiler/rustc_span/src/source_map.rs

```diff
@@ -443,25 +443,39 @@ impl SourceMap {
         sp: Span,
         filename_display_pref: FileNameDisplayPreference,
     ) -> String {
-        if self.files.borrow().source_files.is_empty() || sp.is_dummy() {
-            return "no-location".to_string();
-        }
+        let (source_file, lo_line, lo_col, hi_line, hi_col) = self.span_to_location_info(sp);
+
+        let file_name = match source_file {
+            Some(sf) => sf.name.display(filename_display_pref).to_string(),
+            None => "no-location".to_string(),
```

Comment:
```suggestion
            None => return "no-location".to_string(),
```

otherwise we get `no-location:0:0: 0:0`